src/traces/scatter3d/defaults.js

```diff
@@ -34,6 +34,7 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
 
     coerce('text');
     coerce('mode');
+    coerce('connectgaps');
```

Comment:
This `coerce` call should be in the `hasLines` block below this [line](https://github.com/mikolalysenko/plotly.js/blob/a4c4e0dfa2191861647ad762b494e894db5cc450/src/traces/scatter3d/defaults.js#L40) - as `connectgaps` only makes sense for traces with mode `'lines'`.